packages/terminal/src/widget.ts

```diff
@@ -65,65 +67,48 @@ export class Terminal extends Widget {
       this._session.messageReceived.disconnect(this._onMessage, this);
     }
     this._session = value || null;
-    if (!value) {
-      return;
+    if (value) {
+      this._setSession(value);
     }
-    value.ready.then(() => {
-      if (this.isDisposed || value !== this._session) {
-        return;
-      }
-      value.messageReceived.connect(
-        this._onMessage,
-        this
-      );
-      this.title.label = `Terminal ${value.name}`;
-      this._setSessionSize();
-      if (this._options.initialCommand) {
-        this._session.send({
-          type: 'stdin',
-          content: [this._options.initialCommand + '\r']
-        });
-      }
-    });
   }
 
   /**
-   * Set a config option for the terminal.
+   * Get a config option for the terminal.
    */
-  getOption<K extends keyof Terminal.IOptions>(
+  getOption<K extends keyof ITerminal.IOptions>(
     option: K
-  ): Terminal.IOptions[K] {
+  ): ITerminal.IOptions[K] {
     return this._options[option];
   }
 
   /**
    * Set a config option for the terminal.
    */
-  setOption<K extends keyof Terminal.IOptions>(
+  setOption<K extends keyof ITerminal.IOptions>(
     option: K,
-    value: Terminal.IOptions[K]
+    value: ITerminal.IOptions[K]
   ): void {
     if (this._options[option] === value) {
       return;
     }
 
     this._options[option] = value;
 
-    if (option === 'initialCommand') {
-      return;
-    }
-
-    if (option === 'theme') {
-      if (value === 'light') {
-        this.addClass('jp-mod-light');
-        this._term.setOption('theme', Private.lightTheme);
-      } else {
-        this.removeClass('jp-mod-light');
-        this._term.setOption('theme', Private.darkTheme);
-      }
-    } else {
-      this._term.setOption(option, value);
-      this._needsResize = true;
+    switch (option) {
```

Comment:
seemed reasonable to `switch` this up...

Reply:
Sure. This will conflict with #5964, but we should be able to deal.